internal/service/alarms/internal/serviceconfig/serviceconfig.go

```diff
@@ -0,0 +1,300 @@
+						Name: "postgres-server-passwords",
+					},
+					Key: utils.AlarmsPasswordEnvName,
+				},
+			},
+		},
+		{Name: "PGAPPNAME", Value: cleanupCronJobName},
+		{Name: "PGCONNECT_TIMEOUT", Value: "10"},
+	}
+
+	return batchv1.CronJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cleanupCronJobName,
+			Namespace: utils.DefaultNamespace,
+			Labels:    commonLabels,
+		},
+		Spec: batchv1.CronJobSpec{
+			Schedule:                   cronSchedule,
+			ConcurrencyPolicy:          batchv1.ForbidConcurrent,
+			SuccessfulJobsHistoryLimit: &successLimit,
+			FailedJobsHistoryLimit:     &failureLimit,
+			JobTemplate: batchv1.JobTemplateSpec{
+				Spec: batchv1.JobSpec{
+					BackoffLimit: &jobBackoffLimit,
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							RestartPolicy: corev1.RestartPolicyOnFailure,
+							Containers: []corev1.Container{
+								{
+									Name:    cleanupCronJobName,
+									Image:   pgimage,
+									Command: []string{"psql"},
+									Args:    []string{"-f", fmt.Sprintf("%s/%s", cleanScriptDir, cleanScriptName)},
+									Env:     pgEnv,
+									Resources: corev1.ResourceRequirements{
+										Limits: corev1.ResourceList{
+											corev1.ResourceCPU:    resource.MustParse("100m"),
+											corev1.ResourceMemory: resource.MustParse("128Mi"),
+										},
+									},
+									VolumeMounts: []corev1.VolumeMount{
+										{
+											Name:      configMapName,
+											MountPath: cleanScriptDir,
+										},
+									},
+								},
+							},
+							Volumes: []corev1.Volume{{
+								Name: configMapName,
+								VolumeSource: corev1.VolumeSource{
+									ConfigMap: &corev1.ConfigMapVolumeSource{
+										LocalObjectReference: corev1.LocalObjectReference{
+											Name: configMapName,
+										},
+									},
+								},
+							}},
+						},
+					},
+				},
+			},
+		},
+	}, nil
+}
+
+// getPostgresImage dynamically collect the existing PG image to run psql from
+func getPostgresImage(ctx context.Context, c client.Client) (string, error) {
+	var deploymentList appsv1.DeploymentList
+	if err := c.List(ctx, &deploymentList, client.InNamespace(utils.DefaultNamespace),
+		client.MatchingLabels{"app": "postgres-server"}); err != nil {
+		return "", fmt.Errorf("list deployments: %w", err)
+	}
+
+	if len(deploymentList.Items) != 1 {
+		return "", fmt.Errorf("expected one postgres deployment, found %d", len(deploymentList.Items))
+	}
+
+	deployment := deploymentList.Items[0]
+	for _, container := range deployment.Spec.Template.Spec.Containers {
+		if container.Name == "server" {
+			if container.Image == "" {
+				return "", fmt.Errorf("postgres server container has empty image")
+			}
+			return container.Image, nil
+		}
+	}
+
+	return "", fmt.Errorf("server container not found in postgres deployment %s", deployment.Name)
+}
+
+// CleanupResources call this for cleanup of k8s resources
+func CleanupResources(ctx context.Context, hc client.Client) error {
```

Comment:
Do we need to "cleanup" these resources or can we just do an "upsert" type operation on it since we seem to be immediately recreating them in all cases.

Reply:
Yeah mostly it was to force cronjob to quickly read the update configmap. But thinking about it again now that you mention it....if we do update the configmap...all the subsequent pods will read the latest cm anyway...argh yeah will make this cleaner.

Reply:
cleaned it up and dropped the delete step